python_http_client/exceptions.py

```diff
@@ -0,0 +1,53 @@
+class HTTPError(Exception):
+	''' Base of all other errors'''
+	def __init__(self,error):
+		self.code = error.code
+		self.reason = error.reason
+		#self.headers = error.headers
+
+class BadRequestsError(HTTPError):
+	pass
+
+class UnauthorizedError(HTTPError):
+	pass
+
+class ForbiddenError(HTTPError):
+	pass
+
+class NotFoundError(HTTPError):
+	pass
+
+class MethodNotAllowedError(HTTPError):
+	pass
+
+class PayloadTooLargeError(HTTPError):
+	pass
+
+class UnsupportedMediaTypeError(HTTPError):
+	pass
+
+class TooManyRequestsError(HTTPError):
+	pass
+
+class InternalServerError(HTTPError):
+	pass
+
+class ServiceUnavailableError(HTTPError):
+	pass
+
+err_dict = { 	400 : BadRequestsError,
+				401 : UnauthorizedError,
+				403 : ForbiddenError,
+				404 : NotFoundError,
+				405 : MethodNotAllowedError,
+				413 : PayloadTooLargeError,
+				415 : UnsupportedMediaTypeError,
+				429 : TooManyRequestsError,
+				500 : InternalServerError,
+				503 : ServiceUnavailableError
+}
+
+def handle_error(error):
+	exc = err_dict[error.code](error)
+	exc.__cause__ = None
```

Comment:
If we won't to do this then an ugly traceback is produced in Python 3. In this PR, we are raising our custom exception when an urllib.HTTPError is raised. Here, in python 3 , such a thing will be displayed the output.
```
Traceback (most recent call last):
  File "/home/dibya/workspace/python-http-client/python_http_client/client.py", line 141, in _make_request
    return opener.open(request)
  File "/usr/lib/python3.5/urllib/request.py", line 472, in open
    response = meth(req, response)
  File "/usr/lib/python3.5/urllib/request.py", line 582, in http_response
    'http', request, response, code, msg, hdrs)
  File "/usr/lib/python3.5/urllib/request.py", line 504, in error
    result = self._call_chain(*args)
  File "/usr/lib/python3.5/urllib/request.py", line 444, in _call_chain
    result = func(*args)
  File "/usr/lib/python3.5/urllib/request.py", line 696, in http_error_302
    return self.parent.open(new, timeout=req.timeout)
  File "/usr/lib/python3.5/urllib/request.py", line 472, in open
    response = meth(req, response)
  File "/usr/lib/python3.5/urllib/request.py", line 582, in http_response
    'http', request, response, code, msg, hdrs)
  File "/usr/lib/python3.5/urllib/request.py", line 510, in error
    return self._call_chain(*args)
  File "/usr/lib/python3.5/urllib/request.py", line 444, in _call_chain
    result = func(*args)
  File "/usr/lib/python3.5/urllib/request.py", line 590, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 401: UNAUTHORIZED

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "test.py", line 7, in <module>
    r = a.name.api.get()
  File "/home/dibya/workspace/python-http-client/python_http_client/client.py", line 209, in http_request
    return Response(self._make_request(opener, request))
  File "/home/dibya/workspace/python-http-client/python_http_client/client.py", line 143, in _make_request
    handle_error(err)
  File "/home/dibya/workspace/python-http-client/python_http_client/exceptions.py", line 53, in handle_error
    raise exc
python_http_client.exceptions.UnauthorizedError: HTTP Error 401: UNAUTHORIZED

```
If we set the `__cause__` to None, then only the part below `During handling of the above exception, another exception occurred:` is shown, which is all that should be displayed.